sympy/core/power.py

```diff
@@ -487,6 +487,20 @@ def _eval_is_finite(self):
             if self.exp.is_nonnegative or self.base.is_nonzero:
                 return True
 
+    def _eval_is_prime(self):
+        if self.exp == S.One:
+            return self.base.is_prime
+        if self.is_number:
+            return self.doit().is_prime
+
+        if self.is_integer and self.is_positive:
+            """
+            a Power will be non-prime only if both base and exponent
+            are greater than 1
+            """
+            if (self.base-1).is_positive or (self.exp-1).is_positive:
```

Comment:
> `exp(log(3)) == 3` is prime.

`exp(log(3))` is an instance of `sympy.functions.elementary.exponential.exp`.
So nothing really can be done here.

Use of any special functions will lead to an object of its own class.
Using explicit values result in a `Float` number which cannot be evaluated as prime.

> There need to be some checks if they are integers as well.

Control reaches here only if the power can be evaluated as an integer